File: patterns/manuelle-trigger-erkennung/rules-python/manuelle_trigger_erkennung_deadman.py
from openhab import rule, Registry
from openhab.triggers import GroupStateUpdateTrigger, SystemStartlevelTrigger, ItemCommandTrigger


@rule(triggers=[SystemStartlevelTrigger(100)])
class SystemStarted:
    def execute(self, module, input):
        Registry.getItem("DeadMansSwitch").sendCommand("STARTUP")


@rule(triggers=[ItemCommandTrigger("SomeRuleTrigger", "ON")])
class RuleThatChangesAGWatchItem:
    def execute(self, module, input):
        Registry.getItem("DeadMansSwitch").sendCommand("RULE")
        # Aktionen ausfuehren
        Registry.getItem("WatchedItem1").sendCommand("ON")
        Registry.getItem("DeadMansSwitch").sendCommand("MANUAL")


@rule(triggers=[GroupStateUpdateTrigger("gWatchItems")])
class IsManuallyTriggered:
    def execute(self, module, input):
        if str(Registry.getItem("DeadMansSwitch").getState()) == "MANUAL":
            self.logger.info("Element wurde manuell ausgeloest")
        else:
            self.logger.info("Element wurde durch eine Regel ausgeloest (DeadMansSwitch=RULE)")

File: patterns/manuelle-trigger-erkennung/rules-js/manuelle-trigger-erkennung-deadman.js
rules.JSRule({
  name: "System Started",
  triggers: [triggers.SystemStartlevelTrigger(100)],
  execute: (event) => {
    items.getItem("DeadMansSwitch").sendCommand("STARTUP");
  }
});

rules.JSRule({
  name: "Rule that changes a gWatchItem",
  triggers: [triggers.ItemCommandTrigger("SomeRuleTrigger", "ON")],
  execute: (event) => {
    items.getItem("DeadMansSwitch").sendCommand("RULE");
    // Aktionen ausfuehren
    items.getItem("WatchedItem1").sendCommand("ON");
    items.getItem("DeadMansSwitch").sendCommand("MANUAL");
  }
});

rules.JSRule({
  name: "Is Manually Triggered?",
  triggers: [triggers.GroupStateUpdateTrigger("gWatchItems")],
  execute: (event) => {
    if (items.getItem("DeadMansSwitch").state === "MANUAL") {
      console.log("Element wurde manuell ausgeloest");
    } else {
      console.log("Element wurde durch eine Regel ausgeloest (DeadMansSwitch=RULE)");
    }
  }
});
